File: lib/Direction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isOrientationIdent = (ident, plane) => Object.keys(plane).indexOf(ident) !== -1;
exports.fromPoint = (point, plane) => {
    const ident = Object.keys(plane).filter((dir) => plane[dir].delta.equals(point))[0];
    if (exports.isOrientationIdent(ident, plane) === true) {
        return plane[ident];
    }
    else {
        return undefined;
    }
};

File: src/Direction.py
from namedlist import namedlist

from .Point import Point

Direction = namedlist('Direction', ['delta', 'left', 'right'])
Orientation = namedlist('Orientation', ['up', 'right', 'down', 'left'])

"""
point: Point
plane: Orientation
"""
def fromPoint(point, plane):
  for direction in plane._asdict().values():
    if direction.delta == point:
      return direction

  return None